intranet/middleware/access_log.py

```diff
@@ -27,9 +29,16 @@ def __call__(self, request):
         if isinstance(ip, set):
             ip = ip[0]
 
-        log_line = "{} - {} - [{}] \"{}\" \"{}\"".format(ip, username, datetime.now(), request.get_full_path(), request.META.get(
-            "HTTP_USER_AGENT", ""))
+        user_agent = request.META.get("HTTP_USER_AGENT", "")
+        log_line = '{} - {} - [{}] "{}" "{}"'.format(ip, username, datetime.now(), request.get_full_path(), user_agent)
+
+        loggable = True
+        for user_agent_substring in settings.NONLOGGABLE_USER_AGENT_SUBSTRINGS:
```

Comment:
1. I would wrap this in an `if user_agent:` in case the user agent header is empty.
2. This whole logic just begs to be shortened to `if not any(user_agent_substring in user_agent for user_agent_substring in settings.NONLOGGABLE_USER_AGENT_SUBSTRINGS)`.